x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/common/AdjustableCapacityBlockingQueue.java

```diff
@@ -0,0 +1,119 @@
+/*
```

Comment:
The class contains the majority of the needed functionality to support an adjustable queue. This wraps a blocking queue by adding the overhead of using an `ReentrantReadWriteLock`. I tried thinking of another way to do this without leveraging a lock but it'd require us to shutdown the whole queuing service which seemed worse. I'm open to other ideas though.

Doing it this way allows us to change the capacity without having to restart the node.

I tried to follow what `LinkedBlockingQueue` does in terms of whether to use an `lockInterruptibly()`  call.